File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n+1];
        for(int i=0;i<=n;i++){
            ans[i]=totalSetBitsOf(i);
        }
        return ans;
    }
    public int totalSetBitsOf(int i){
        int ans=0;
        while(i!=0){
            if((i&1)==1){
                ans++;
            }
            i = i >>1;
        }
        return ans;
    }
}

File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>ans;
        for(int i=0;i<=n;i++){
            int c=0;
            int k=i;
            while(k>0){
            c += k&1;
            k=k>>1;
          }
          ans.push_back(c);
        }
        return ans;
    }
};